winpr/libwinpr/crt/string.c

```diff
@@ -98,17 +99,39 @@ int _strnicmp(const char* string1, const char* string2, size_t count)
 
 int _wcscmp(const WCHAR* string1, const WCHAR* string2)
 {
-	WCHAR value1, value2;
+	WINPR_ASSERT(string1);
+	WINPR_ASSERT(string2);
 
-	while (*string1 && (*string1 == *string2))
+	while (*string1 && (*string2))
```

Comment:
Why not just add a check for `*string2` to the old loop. With these changes we now check each string twice for `\0` each iteration

Reply:
replaced with a `while(TRUE)` and the `\0` checks below.